### PR TITLE
DOCS-6436 Add Local and Remote Testing Documentation

### PR DESCRIPTION
**Pull request by alai97**
### What does this PR do? What is the motivation?

Adds a folder of documentation for Local/Remote Testing with three child pages in the Continuous Testing doc set.

Waiting for new illustrations to replace the Testing Tunnel diagrams.

### Merge instructions

- [x] Please merge after reviewing

### CODE REVIEW

File: content/en/continuous_testing/environments/_index.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+## Testing in multiple environments
+
+Continuous Testing can reuse the same scenario from scheduled tests used against the production environment to test publicly available pre-production environments.
+
+Whether it's for a [blue/green deployment][3], or a dedicated staging environment, Continuous Testing allows you to reroute an existing scenario to a different environment using the `startUrlSubstitutionRegex`<!-- and `resourceUrlSubstitutionRegex`--> field.
+
+For more information, see [Testing Multiple Environments][4].
+
+### Testing while using proxies, firewalls, or VPNs
```
Comment (alai97) [inline content/en/continuous_testing/environments/_index.md:45]:
```suggestion
## Testing while using proxies, firewalls, or VPNs
```

File: content/en/continuous_testing/environments/proxy_firewall_vpn.md [hunk 2]
```diff
@@ -1,65 +1,60 @@
...
+Datadog recommends using the testing tunnel if you need to launch Continuous Testing tests against local versions of your application without deploying a dedicated, long-lasting probing system such as a [private location][1]. You can also use the testing tunnel to trigger tests on short-lived cloud environments.
 
 ## What is the testing tunnel?
 
-The testing tunnel is a functionality that comes with the [@datadog/datadog-ci][2] NPM package which is one of the methods Datadog <span class="x x-first x-last">provides </span>to include your Synthetic tests as part of your CI/CD pipelines. The testing tunnel creates an end-to-end encrypted HTTP proxy between your infrastructure and Datadog, meaning that any test requests sent <span class="x x-first x-last">through</span> the CLI are automatically routed through the `datadog-ci` client<span class="x x-first x-last">. This allows</span> Datadog to access and test your internal applications.
+The testing tunnel is a functionality that comes with the [@datadog/datadog-ci][2] NPM package which is one of the methods Datadog provides to include your Synthetic tests as part of your CI/CD pipelines.
```
Comment (jakepruitt) [inline content/en/continuous_testing/environments/proxy_firewall_vpn.md:25]:
```suggestion
The testing tunnel is a functionality that comes with the [@datadog/datadog-ci][2] NPM package, which is one of the methods Datadog provides to include your Synthetic tests as part of your CI/CD pipelines.
```

**Review by jakepruitt — APPROVED** [review]

File: content/en/continuous_testing/environments/_index.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+- link: "https://learn.datadoghq.com/courses/synthetic-tests-ci-cd-pipeline"
+  tag: "Learning Center"
+  text: "Learn how to run tests in a CI/CD pipeline"
+- link: "/continuous_testing/environments/multiple_env"
+  tag: "Documentation"
+  text: "Learn about testing in multiple environments"
+- link: "/continuous_testing/environments/proxy_firewall_vpn"
+  tag: "Documentation"
+  text: "Learn about testing while using proxies, firewalls, or VPNs"
+- text: "/synthetics/private_locations"
```
Comment (drichards-87) [inline content/en/continuous_testing/environments/_index.md:24]:
```suggestion
- link: "/synthetics/private_locations"
```

File: content/en/synthetics/browser_tests/_index.md [hunk 4]
```diff
@@ -209,6 +209,6 @@ You can restrict access to a browser test based on the roles in your organizatio
 [12]: /synthetics/browser_tests/actions/
 [13]: /account_management/rbac#custom-roles
 [14]: /account_management/rbac/#create-a-custom-role
-[15]: /continuous_testing/testing_tunnel
+[15]: /continuous_testing/proxy_firewall_vpn
```
Comment (drichards-87) [inline content/en/synthetics/browser_tests/_index.md:212]:
```suggestion
[15]: /continuous_testing/environments/proxy_firewall_vpn
```

File: content/en/continuous_testing/environments/proxy_firewall_vpn.md [hunk 5]
```diff
@@ -1,65 +1,60 @@
 ---
-title: Continuous Testing Tunnel
+title: Testing While Using Proxies, Firewalls, or VPNs
```
Comment (drichards-87) [inline content/en/continuous_testing/environments/proxy_firewall_vpn.md:2]:
```suggestion
title: Testing with Proxies, Firewalls, or VPNs
```

File: content/en/continuous_testing/environments/multiple_env.md [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+
+This option allows you to reuse the same test scenario on both the production environment and any development environments (such as staging) as long as they are publicly available. If some of your development environments are private, see [Testing While Using Proxies, Firewalls, or VPNs][3] to learn how to test against [private environments][4].
+
+## Partially modifying the starting URL
+
+If some of your tests start at the homepage, or a similarly simple URL, the previous solution works fine, but it doesn't cover every use case. Blindly replacing the starting URL may unintentionally remove the path or some search query parameters from the URL that the scenario is expected to test.
+
+In addition to `startUrl`, the `startUrlSubstitutionRegex` field allows you to modify the starting URL without overwriting it entirely. This option allows you to substitute parts of the default starting URL based on the provided regular expression.
+
+This field expects a string containing two parts, separated by a pipe character `|`: `<regex>|<rewritting rule>`. The first part is the regex to apply to the default starting URL. The second is the expression to rewrite the URL.
```
Comment (drichards-87) [inline content/en/continuous_testing/environments/multiple_env.md:48]:
```suggestion
This field expects a string containing two parts, separated by a pipe character `|`: `<regex>|<rewriting rule>`. The first part is the regex to apply to the default starting URL. The second is the expression to rewrite the URL.
```

File: content/en/continuous_testing/environments/multiple_env.md [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+```shell
+yarn datadog-ci synthetics run-tests --public-id <public-id> --start-url "https://staging.my-app.com"
+```
+
+<!--
+Note: I'm not sure about that.
+This option also supports using environment variables to allow picking up the URL from the development environment that might expose it as environment variable: `--start-url "https://$DEPLOYMENT_PREFIX.my-app.com"`.
+-->
+
+This option allows you to reuse the same test scenario on both the production environment and any development environments (such as staging) as long as they are publicly available. If some of your development environments are private, see [Testing While Using Proxies, Firewalls, or VPNs][3] to learn how to test against [private environments][4].
```
Comment (drichards-87) [inline content/en/continuous_testing/environments/multiple_env.md:40]:
```suggestion
This option allows you to reuse the same test scenario on both the production environment and other development environments (such as staging) as long as they are publicly available. To learn how to test against [private environments][4], see [Testing While Using Proxies, Firewalls, or VPNs][3].
```

File: content/en/continuous_testing/environments/multiple_env.md [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+
+## Partially modifying the starting URL
+
+If some of your tests start at the homepage, or a similarly simple URL, the previous solution works fine, but it doesn't cover every use case. Blindly replacing the starting URL may unintentionally remove the path or some search query parameters from the URL that the scenario is expected to test.
```
Comment (drichards-87) [inline content/en/continuous_testing/environments/multiple_env.md:44]:
```suggestion
If some of your tests start at the homepage, or a similarly simple URL, the previous solution works fine, but it doesn't cover every use case. Blindly replacing the starting URL may unintentionally remove the path or certain search query parameters from the URL that the scenario is expected to test.
```

**Review by drichards-87 — APPROVED** [review]
Left you some suggestions and approved the PR.

File: content/en/continuous_testing/environments/_index.md [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+  tag: "Documentation"
+  text: "Learn about Private Locations"
+---
+
+## Overview
+
+In the context of [testing within a CI/CD pipeline, also known as shift-left testing][1], the production environment is typically the last link in the chain. Your application is likely to go through a number of steps before reaching this stage.
```
Comment (alai97) [inline content/en/continuous_testing/environments/_index.md:31]:
```suggestion
In the context of [testing within a CI/CD pipeline, also known as shift-left testing][1], the production environment is typically the last link in the chain. Your application is likely to go through several steps before reaching this stage.
```

File: content/en/continuous_testing/environments/multiple_env.md [hunk 10]
```diff
@@ -0,0 +1,122 @@
...
+When triggering a CI test, the `startUrl` field allows you to overwrite the first URL that a browser test navigates to or the URL used by an HTTP test request. You can specify this option through the global configuration file, the Synthetics configuration files (`*.synthetics.json`), or the command line flag `--start-url`.
+
+```shell
+yarn datadog-ci synthetics run-tests --public-id <public-id> --start-url "https://staging.my-app.com"
+```
+
+<!--
+Note: I'm not sure about that.
+This option also supports using environment variables to allow picking up the URL from the development environment that might expose it as environment variable: `--start-url "https://$DEPLOYMENT_PREFIX.my-app.com"`.
+-->
```
Comment (alai97) [inline content/en/continuous_testing/environments/multiple_env.md:38]:
```suggestion
```

File: content/en/developers/ide_integrations/vscode/_index.md [hunk 11]
```diff
@@ -7,7 +7,7 @@ further_reading:
 - link: "/getting_started/synthetics/"
   tag: "Documentation"
   text: "Getting Started with Synthetic Monitoring"
-- link: "/continuous_testing/testing_tunnel/"
+- link: "/continuous_testing/"
   tag: "Documentation"
   text: "Learn about the Continuous Testing Tunnel"
```
Comment (alai97) [inline content/en/developers/ide_integrations/vscode/_index.md:12]:
```suggestion
  text: "Learn about Continuous Testing"
```

File: content/en/continuous_testing/_index.md [hunk 12]
```diff
@@ -56,19 +61,19 @@ You can use the [Datadog Terraform provider][10] to control test creation and st
 
 ## Accelerate troubleshooting
 
-Performing tests in a unified monitoring platform helps you quickly find the root cause of failed test runs and reduce MTTR. You can obtain the full context for troubleshooting—without switching between tools—through correlated metrics, traces, and logs surfaced by the Datadog [APM integration][12] by looking at executed jobs in the [Synthetic Monitoring & Continuous Testing Explorer][11]. 
+Performing tests in a unified monitoring platform helps you quickly find the root cause of failed test runs and reduce MTTR. You can obtain the full context for troubleshooting—without switching between tools—through correlated metrics, traces, and logs surfaced by the Datadog [APM integration][12] by looking at executed jobs in the [Synthetic Monitoring & Continuous Testing Explorer][11].
```
Comment (alai97) [inline content/en/continuous_testing/_index.md:64]:
```suggestion
Performing tests in a unified monitoring platform helps you find the root cause of failed test runs and reduce MTTR. You can obtain the full context for troubleshooting—without switching between tools—through correlated metrics, traces, and logs surfaced by the Datadog [APM integration][12] by looking at executed jobs in the [Synthetic Monitoring & Continuous Testing Explorer][11].
```

File: content/en/continuous_testing/environments/multiple_env.md [hunk 13]
```diff
@@ -0,0 +1,118 @@
...
+## Overriding the starting URL
+
+A Synthetic browser test starts the test scenario by navigating to a starting URL. Similarly, an API HTTP test sends a request to a specific URL. When triggering a CI test, you can overwrite this starting URL to point to another environment where your application is deployed in.
+
+{{< img src="continuous_testing/continuous_testing_start-url_substitution.png" alt="Continuous Testing tunnel allows the Synthetics Worker to reach your private applications" width="100%" >}}
+
+When triggering a CI test, the `startUrl` field allows you to overwrite the first URL that a browser test navigates to or the URL used by an HTTP test request. You can specify this option through the global configuration file, the Synthetics configuration files (`*.synthetics.json`), or the command line flag `--start-url`.
+
+```shell
+yarn datadog-ci synthetics run-tests --public-id <public-id> --start-url "https://staging.my-app.com"
```
Comment (Drarig29) [inline content/en/continuous_testing/environments/multiple_env.md:32]: Not everyone uses `yarn`, and the CLI could also be globally installed. I think what's important is the command line, not how `datadog-ci` is access.

One could totally run it like `/path/to/datadog-ci <args>` as well

```suggestion
datadog-ci synthetics run-tests --public-id <public-id> --start-url "https://staging.my-app.com"
```

Comment (Drarig29) [reply content/en/continuous_testing/environments/multiple_env.md:32]: Can you make sure we don't have `yarn datadog-ci` anywhere else?

File: content/en/continuous_testing/environments/multiple_env.md [hunk 14]
```diff
@@ -0,0 +1,118 @@
...
+
+In addition to `startUrl`, the `startUrlSubstitutionRegex` field allows you to modify the starting URL without overwriting it entirely. This option allows you to substitute parts of the default starting URL based on the provided regular expression.
+
+This field expects a string containing two parts, separated by a pipe character `|`: `<regex>|<rewriting rule>`. The first part is the regex to apply to the default starting URL. The second is the expression to rewrite the URL.
+
+A simple example looks like the following:
+
+```shell
+https://prod.my-app.com/(.*)|https://staging.my-app.com/$1
+```
```
Comment (Drarig29) [inline content/en/continuous_testing/environments/multiple_env.md:50]: Can you remove the `shell` annotation? Apparently the documentation preprocessor adds a visual `$` at the beginning to make people understand it's  a command line. But here it's misleading:

image

File: content/en/continuous_testing/environments/proxy_firewall_vpn.md [hunk 15]
```diff
@@ -1,65 +1,60 @@
...
+First, `datadog-ci` gets a pre-signed URL from Datadog for authentication. Then, it opens a WebSocket secure (WSS) connection to Datadog's managed locations using the pre-signed URL. By using SSH connections through the WebSocket connection, tests are triggered by `datadog-ci` and executed through Datadog's managed locations.
+
+Because DNS resolution is performed through the testing tunnel, you can test applications with internal domains or even on the `localhost` of the machine running `datadog-ci`.
 
 When using the testing tunnel, your tests' locations are overridden by a location that depends on your Datadog account region.
 
 ## How to use the testing tunnel
 
-As mentioned above, the testing tunnel comes with the [@datadog/datadog-ci][2] NPM package and is available from version [v0.11.0][3] of the package. To get started, see [Continuous Testing and CI/CD][4].
+As mentioned above, the testing tunnel comes with the [`@datadog/datadog-ci`][2] NPM package and is available from version [v0.11.0][3] of the package. To get started, see [Continuous Testing and CI/CD][4].
```
Comment (Drarig29) [inline content/en/continuous_testing/environments/proxy_firewall_vpn.md:39]: I think the link to that version isn't really useful, and could mislead users into using this version explicitly, which is very outdated.

Also "available from version" could be misinterpreted as "only available from the version `v0.11.0` specifically". That's a bit farfetched, but I think something like `>=`, or "available **starting** from" would be more explicit 😁 

```suggestion
As mentioned above, the testing tunnel comes with the [`@datadog/datadog-ci`][2] NPM package and is available in versions `>=v0.11.0` of the package. To get started, see [Continuous Testing and CI/CD][4].
```

**Pull request merged**